controller/gacha/gacha.go

```diff
@@ -0,0 +1,157 @@
+package gacha
+
+import (
+	"CATechDojo/controller/request"
+	"CATechDojo/controller/response"
+	"CATechDojo/controller/user"
+	"CATechDojo/model/gacha"
+	"bytes"
+	"encoding/json"
+	"io"
+	"log"
+	"math/rand"
+	"net/http"
+	"time"
+)
+
+func DrawSpecificCharacter(w http.ResponseWriter, r *http.Request) {
+	//x-tokenを受け取る
+	token := r.Header.Get("x-token")
+	if token == "" {
+		log.Println("トークンの値がnilです")
+		http.Error(w, "認証情報が必要です。", http.StatusBadRequest)
+		return
+	}
+
+	//特定のキャラクターをDBから取り出す
+	g := gacha.New()
+
+	if err := g.SelectCharacter(); err != nil {
+		log.Println(err)
+		http.Error(w, "データを参照できませんでした", http.StatusInternalServerError)
+	}
+
+	//取り出したキャラクターをDBに保存する
+	userCharacterID, err := user.CreateUUID()
+	if err != nil {
+		log.Println(err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+
+	if err := g.InsertCharacter(userCharacterID, token); err != nil {
+		log.Println(err)
+		http.Error(w, "ユーザデータを保存できませんでした", http.StatusInternalServerError)
+	}
+
+	//取り出したキャラクターのidとnameをjson形式で返す
+	res := g.GetCharacterData()
+
+	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+	w.WriteHeader(http.StatusOK)
+	data, err := json.Marshal(res)
+	if err != nil {
+		log.Println(err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+	_, _ = w.Write(data)
+}
+
+func Draw(w http.ResponseWriter, r *http.Request) {
+	token := r.Header.Get("x-token")
+	if token == "" {
+		log.Println("トークンの値がnilです")
+		http.Error(w, "認証情報が必要です。", http.StatusBadRequest)
+		return
+	}
+
+	body := r.Body
+	defer body.Close()
+
+	buf := new(bytes.Buffer)
+	if _, err := io.Copy(buf, body); err != nil {
+		log.Println(err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+
+	var times request.Times
+	if err := json.Unmarshal(buf.Bytes(), &times); err != nil {
+		log.Println(err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+
+	//gacha_oddsテーブルから全件取得
+	g := gacha.New()
+
+	odds, err := g.SelectAllOdds()
+	if err != nil {
+		http.Error(w, "データを参照できませんでした", http.StatusInternalServerError)
+	}
+
+	var hitCharacterID string
+	hitCharactersData := make([]gacha.GachaData, 0)
+
+	rand.Seed(time.Now().UnixNano())
+
+	for i := 1; i <= times.Times; i++ {
```

Comment:
`i=1`とするのは結構トリッキーなので，`i=0`から始まるfor文に修正しよう！

```suggestion
	for i := 0; i < times.Times; i++ {
```

Reply:
@sne-5249

Reply:
はい！修正しました。